packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -372,6 +442,192 @@ describe("Presence", () => {
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Disconnected,
+								"Stale attendee should be 'Disconnected' 30s after reconnection",
+							);
+							assert.strictEqual(
+								disconnectedAttendees.length,
+								1,
+								"Exactly one attendee should be announced as disconnected",
+							);
+						});
+
+						it.skip("does not update stale attendee status if local client does not reconnect", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect local client and wait 60s
+							runtime.disconnect();
+							clock.tick(60_000);
+
+							// Verify - stale attendee should still be 'Connected' after 30+ seconds
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Stale attendee should still be 'Connected' after 30s",
+							);
+						});
+
+						it.skip("does not update stale attendee status if local client reconnection lasts less than 30s", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect, reconnect for 15 second, then disconnect local client again
+							runtime.disconnect(); // First disconnect
+							clock.tick(1000);
+							runtime.connect("client6"); // Reconnect
+							clock.tick(15_000); // Advance 15 seconds
+							runtime.disconnect(); // Disconnect again
+							clock.tick(60_000); // Advance 60 seconds
+
+							// Verify - stale attendee should still be 'Connected' after 30+ seconds post-reconnection
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Stale attendee should still be 'Connected' after 30s",
+							);
+						});
+
+						it.skip("does not update attendee status to 'Disconnected' if stale attendee rejoins", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect, reconnect, then process rejoin signal from known attendee after 15s
+							runtime.disconnect();
+							clock.tick(1000);
+							runtime.connect("client6");
+							clock.tick(15_000);
+							const joinedAttendees = processJoinSignals([rejoinAttendeeSignal]);
+							clock.tick(60_000);
+
+							// Verify - rejoining attendee should still be 'Connected' with no `attendeeJoined` announced 30+ seconds post-reconnection
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Active attendee should still be 'Connected' 30s after reconnection",
+							);
+							assert.strictEqual(
+								joinedAttendees.length,
+								0,
+								"No `attendeeJoined` should be announced for rejoining attendee that's already 'Connected'",
+							);
+						});
+
+						it.skip("does not update attendee status to 'Disconnected' if stale attendee sends datastore update", () => {
```

Comment:
These two are very similar. Yet they verify different things. Can/should be the same, right?

Reply:
Looking at this also inspired another test case for when an attendee with a stale connection becomes active by when their connection is removed. We should mark them as active, so we don't end up announcing `attendeeDisconnected` twice after 30s.